server/src/main/java/org/activityinfo/core/shared/util/MimeTypeUtil.java

```diff
@@ -48,198 +49,224 @@
-//        pgm	image/x-portable-graymap
-//        pko	application/ynd.ms-pkipko
-//        pma	application/x-perfmon
-//        pmc	application/x-perfmon
-//        pml	application/x-perfmon
-//        pmr	application/x-perfmon
-//        pmw	application/x-perfmon
-//        pnm	image/x-portable-anymap
-//        pot	application/vnd.ms-powerpoint
-//        ppm	image/x-portable-pixmap
-//        pps	application/vnd.ms-powerpoint
-//        ppt	application/vnd.ms-powerpoint
-//        prf	application/pics-rules
-//        ps	application/postscript
-//        pub	application/x-mspublisher
-//        qt	video/quicktime
-//        ra	audio/x-pn-realaudio
-//        ram	audio/x-pn-realaudio
-//        ras	image/x-cmu-raster
-//        rgb	image/x-rgb
-//        rmi	audio/mid
-//        roff	application/x-troff
-//        rtf	application/rtf
-//        rtx	text/richtext
-//        scd	application/x-msschedule
-//        sct	text/scriptlet
-//        setpay	application/set-payment-initiation
-//        setreg	application/set-registration-initiation
-//        sh	application/x-sh
-//        shar	application/x-shar
-//        sit	application/x-stuffit
-//        snd	audio/basic
-//        spc	application/x-pkcs7-certificates
-//        spl	application/futuresplash
-//        src	application/x-wais-source
-//        sst	application/vnd.ms-pkicertstore
-//        stl	application/vnd.ms-pkistl
-//        stm	text/html
-//        sv4cpio	application/x-sv4cpio
-//        sv4crc	application/x-sv4crc
-//        svg	image/svg+xml
-//        swf	application/x-shockwave-flash
-//        t	application/x-troff
-//        tar	application/x-tar
-//        tcl	application/x-tcl
-//        tex	application/x-tex
-//        texi	application/x-texinfo
-//        texinfo	application/x-texinfo
-//        tgz	application/x-compressed
-//        tif	image/tiff
-//        tiff	image/tiff
-//        tr	application/x-troff
-//        trm	application/x-msterminal
-//        tsv	text/tab-separated-values
-//        uls	text/iuls
-//        ustar	application/x-ustar
-//        vcf	text/x-vcard
-//        vrml	x-world/x-vrml
-//        wav audio/x-wav
-//        wcm	application/vnd.ms-works
-//        wdb	application/vnd.ms-works
-//        wks	application/vnd.ms-works
-//        wmf	application/x-msmetafile
-//        wps	application/vnd.ms-works
-//        wri	application/x-mswrite
-//        wrl	x-world/x-vrml
-//        wrz	x-world/x-vrml
-//        xaf	x-world/x-vrml
-//        xbm	image/x-xbitmap
-//        xla	application/vnd.ms-excel
-//        xlc	application/vnd.ms-excel
-//        xlm	application/vnd.ms-excel
-//        xls	application/vnd.ms-excel
-//        xlt	application/vnd.ms-excel
-//        xlw	application/vnd.ms-excel
-//        xof	x-world/x-vrml
-//        xpm	image/x-xpixmap
-//        xwd	image/x-xwindowdump
-//        z	application/x-compress
-//        zip	application/zip
+        fileExtensionToMimeTypeRegistry.put("txt", "text/plain");
```

Comment:
Not sure how to easily workaround it. We identify mimeType on client and put in `FieldValue`. (Which later goes to `FormInstance` if save is clicked). `MimeType` is not used at all during upload (it's used only on download). When we are intermediate state when upload is made but `FormInstance` is not saved then that mime-type is only on client.